openapi3/refs.go

```diff
@@ -107,11 +139,22 @@ func (value *ParameterRef) Validate(c context.Context) error {
 	return v.Validate(c)
 }
 
+func (value ParameterRef) JSONLookup(token string) (interface{}, error) {
+	if token == "$ref" {
+		return value.Ref, nil
+	}
+
+	ptr, _, err := jsonpointer.GetForToken(value.Value, token)
+	return ptr, err
+}
+
 type ResponseRef struct {
 	Ref   string
 	Value *Response
 }
 
+var _ jsonpointer.JSONPointable = (*ResponseRef)(nil)
```

Comment:
Added the compile-time checks, I think that I actually left that in accidentally, but it's certainly not bad to have compile-time contracts.
I don't think runtime tests to ensure specific types implement the interface makes sense, however. The compile-time contracts ensure that already.

> What's your heuristic to know which types should be taken into account?

Generally, anything that is refable, or contains a refable type needs a JSONLookup implementation. 

Refable types need to be able to return a `Ref` type or the real type from `.Value` depending on the situation

But then thanks to `jsonpointer` quirks: Things that contain refable types need a JSONLookup implementation as they will be the last JSONLookup call before returning the value. Which means there is no call on the value struct (which may be a *Ref struct, which needs to return a Ref value or the real type). 

---
Of note, if I were to suggest a comprehensive test to ensure jsonpointer support, it would be an openapi file, along with a series of jsonpointers that reference every openapi type and ensure that the pointer returns the correct marshalled YAML. However, that's far out of the bandwidth that I can afford to spend on this unfortunately.

Reply:
> Generally, anything that is refable, or contains a refable type needs a JSONLookup implementation.

That's the test I'm asking for: use "reflect" to go from the openapi3.Swagger type down through its innards to ensure everything that matches your quoted sentence implements the interface. A much shorter test than what you have in mind, a better chance at being exhaustive as well.
We can add this test in a follow up PR.